xarray/tests/test_dataarray.py

```diff
@@ -1836,100 +1836,249 @@ def test_resample(self):
+            array[[2, 0, 1]].resample(time='1D')
 
     def test_resample_first(self):
         times = pd.date_range('2000-01-01', freq='6H', periods=10)
         array = DataArray(np.arange(10), [('time', times)])
 
-        actual = array.resample('1D', dim='time', how='first')
+        actual = array.resample(time='1D').first()
         expected = DataArray([0, 4, 8], [('time', times[::4])])
         self.assertDataArrayIdentical(expected, actual)
 
         # verify that labels don't use the first value
-        actual = array.resample('24H', dim='time', how='first')
-        expected = DataArray(array.to_series().resample('24H', how='first'))
+        actual = array.resample(time='24H').first()
+        expected = DataArray(array.to_series().resample('24H').first())
         self.assertDataArrayIdentical(expected, actual)
 
         # missing values
         array = array.astype(float)
         array[:2] = np.nan
-        actual = array.resample('1D', dim='time', how='first')
+        actual = array.resample(time='1D').first()
         expected = DataArray([2, 4, 8], [('time', times[::4])])
         self.assertDataArrayIdentical(expected, actual)
 
-        actual = array.resample('1D', dim='time', how='first', skipna=False)
+        actual = array.resample(time='1D').first(skipna=False)
         expected = DataArray([np.nan, 4, 8], [('time', times[::4])])
         self.assertDataArrayIdentical(expected, actual)
 
         # regression test for http://stackoverflow.com/questions/33158558/
         array = Dataset({'time': times})['time']
-        actual = array.resample('1D', dim='time', how='last')
+        actual = array.resample(time='1D').last()
         expected_times = pd.to_datetime(['2000-01-01T18', '2000-01-02T18',
                                          '2000-01-03T06'])
         expected = DataArray(expected_times, [('time', times[::4])],
                              name='time')
         self.assertDataArrayIdentical(expected, actual)
 
-    def test_resample_first_keep_attrs(self):
+    def test_resample_bad_resample_dim(self):
         times = pd.date_range('2000-01-01', freq='6H', periods=10)
-        array = DataArray(np.arange(10), [('time', times)])
-        array.attrs['meta'] = 'data'
-
-        resampled_array = array.resample('1D', dim='time', how='first',
-                                         keep_attrs=True)
-        actual = resampled_array.attrs
-        expected = array.attrs
-        self.assertEqual(expected, actual)
-
-        resampled_array = array.resample('1D', dim='time', how='first',
-                                         keep_attrs=False)
-        assert resampled_array.attrs == {}
-
-    def test_resample_mean_keep_attrs(self):
+        array = DataArray(np.arange(10), [('__resample_dim__', times)])
+        actual = array.resample('1D', dim='__resample_dim__', how='first')
+        self.assertRaisesRegexp(ValueError, 'Proxy resampling dimension')
```

Comment:
I think this needs to be:

```Python
with self.assertRaisesRegexp(ValueError, 'Proxy resampling dimension'):
    actual = array.resample('1D', dim='__resample_dim__', how='first')
```

I'm not sure why this is passing actually.

Reply:
This was butchered in more ways than one; I fixed the test which revealed some underlying flaws, too, which are also fixed.